azurerm/internal/services/network/resource_arm_private_endpoint.go

```diff
@@ -193,7 +198,20 @@ func resourceArmPrivateEndpointRead(d *schema.ResourceData, meta interface{}) er
 	}
 
 	if props := resp.PrivateEndpointProperties; props != nil {
+		privateIpAddress := ""
+
+		if nics := props.NetworkInterfaces; nics != nil && len(*nics) > 0 {
+			nic := (*nics)[0]
+			if nic.ID != nil && *nic.ID != "" {
+				privateIpAddress = getPrivateIpAddress(ctx, nicsClient, *nic.ID)
+			}
+		}
+
 		flattenedConnection := flattenArmPrivateLinkEndpointServiceConnection(props.PrivateLinkServiceConnections, props.ManualPrivateLinkServiceConnections)
+		for _, item := range flattenedConnection {
+			v := item.(map[string]interface{})
+			v["private_ip_address"] = privateIpAddress
+		}
```

Comment:
I don't think you need to loop over the returned slice you just need to append the value to it if the `privateIpAddress` is not empty, something like the below... WDYT?
```
if privateIpAddress != "" {
	privateIp := make(map[string]interface{})
	privateIp["private_ip_address"] = privateIpAddress 
	flattenedConnection = append(flattenedConnection, privateIp)
}
```

Reply:
@WodansSon  the `flattenedConnection` is a list of map, append it with another map will cause `flattenedConnection` has two maps, not a merged map.

Reply:
@njuCZ I got the iteration mixed up in my head... you're right... this is fine.